lib/manageiq/messaging/kafka/queue.rb

```diff
@@ -0,0 +1,19 @@
+module ManageIQ
+  module Messaging
+    module Kafka
+      module Queue
+        def publish_message_impl(_options, &_block)
+          raise "Method Not Implemented"
+        end
+
+        def publish_messages_impl(_messages)
+          raise "Method Not Implemented"
+        end
+
+        def subscribe_messages_impl(_options)
```

Comment:
Hm @bzwei I thought subscribe_messages is what we wanted, from https://github.com/ManageIQ/manageiq-messaging#publish-and-subscribe-messages

> This is the one-to-one publish/subscribe pattern. Multiple subscribers can subscribe to the same queue but only one will consume the message. Subscribers are load balanced.

Topics (https://github.com/ManageIQ/manageiq-messaging#publish-and-subscribe-topics-events) is definitely useful for an event bus but not so much for what we were hoping to use this for first.  If you just haven't gotten to it yet then disregard :)

Reply:
I can add subscribe_messages implementation in a second PR. We will not be able to do call back after message is consumed.